hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElementType.java

```diff
@@ -377,7 +377,12 @@ public String getPropertyTableName(String propertyName) {
 		checkInitialized();
 		if ( this.persister != null ) {
 			AbstractEntityPersister aep = (AbstractEntityPersister) this.persister;
-			return aep.getPropertyTableName( propertyName );
+			try {
+				return aep.getSubclassTableName( aep.getSubclassPropertyTableNumber( propertyName ) );
+			}
+			catch (QueryException e) {
```

Comment:
If this could be avoided, that would be great. Otherwise, please document why this is done.

Reply:
Unfortunate decision in the `AbstractEntityPersister`, it should just have returned `null` and not throw any exception, let alone a completely irrelevant `QueryException`... Its just an indication of property not present in the property mapping / metamodel. This happens for many reasons:  for example "nullable" (not optional, but not-proxyable) association properties are always are missing (require a join), nested id class properties are sometimes missing (Hibernate's metamodel is faulty here - known bug). This try-catch statement is done at several places in Hibernate, and can only avoided by extending the AEP with a method that does precisely this. I think having the try-catch block here and there is preferable to extending the already fairly sophisticated API of the AEP, for remaining compatibility with AEP extensions as much as possible.